cmd/finch-daemon/main.go

```diff
@@ -99,6 +105,24 @@ func run(options *DaemonOptions) error {
 	if err := os.Chown(options.socketAddr, options.socketOwner, options.socketOwner); err != nil {
 		return fmt.Errorf("failed to chown the finch-daemon socket: %w", err)
 	}
+
+	if options.debugAddress != "" {
+		logger.Infof("Serving debugging endpoint on %q", options.debugAddress)
+		go func() {
+			debugListener, err := net.Listen("tcp", options.debugAddress)
+			if err != nil {
+				logger.Fatal(err)
+			}
+			debugServer := &http.Server{
+				Handler:           http.DefaultServeMux,
+				ReadHeaderTimeout: 5 * time.Second,
+			}
+			if err := debugServer.Serve(debugListener); err != nil && !errors.Is(err, http.ErrServerClosed) {
```

Comment:
Can you not use ` http.ListenAndServe(cfg.DebugAddress, nil)` here? https://pkg.go.dev/net/http/pprof

Reply:
Got a gosec warning about how `http.ListenAndServe` doesn't set a timeout by default. I can switch to it if you think its better though.

Reply:
I see. I am not sure if ReadHeaderTimeouts provide much value, since the request may time out even before we receive any requestHeaders. Should we track ReadTimeouts instead and set it to something like 30sec. I am less worried since this is meant to be used in a debug environment.

Reply:
The headers always come before the rest of the request. I think its reasonable to expect a local debugging tool (e.g. curl) to send headers in 5 seconds or less.